pkg/router/controller/route_secret_manager_test.go

```diff
@@ -0,0 +1,1139 @@
+			Allowed: t.allow,
+		},
+	}, t.err
+}
+
+type testSecretGetter struct {
+	namespace string
+	secret    *corev1.Secret
+}
+
+func (t *testSecretGetter) Secrets(_ string) corev1client.SecretInterface {
+	return testclient.NewSimpleClientset(t.secret).CoreV1().Secrets(t.namespace)
+}
+
+// fakeSecretInformer will list/watch only one secret inside a namespace
+func fakeSecretInformer(ctx context.Context, fakeKubeClient *testclient.Clientset, namespace, name string) cache.SharedInformer {
+	fieldSelector := fields.OneTermEqualSelector("metadata.name", name).String()
+	klog.Info(fieldSelector)
+	return cache.NewSharedInformer(
+		&cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				options.FieldSelector = fieldSelector
+				return fakeKubeClient.CoreV1().Secrets(namespace).List(ctx, options)
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				options.FieldSelector = fieldSelector
+				return fakeKubeClient.CoreV1().Secrets(namespace).Watch(ctx, options)
+			},
+		},
+		&corev1.Secret{},
+		0,
+	)
+}
+
+func fakeSecret(namespace, name string, secretType corev1.SecretType, data map[string][]byte) *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Data: data,
+		Type: secretType,
+	}
+}
+
+type fakePluginNew struct {
+	t      watch.EventType
+	route  *routev1.Route
+	err    error
+	stopCh chan struct{}
+}
+
+func (p *fakePluginNew) HandleRoute(t watch.EventType, route *routev1.Route) error {
+	defer close(p.stopCh)
+	p.t, p.route = t, route
+	return p.err
+}
+
+func (p *fakePluginNew) HandleNode(t watch.EventType, node *corev1.Node) error {
+	return fmt.Errorf("not expected")
+}
+
+func (p *fakePluginNew) HandleEndpoints(watch.EventType, *corev1.Endpoints) error {
+	return fmt.Errorf("not expected")
+}
+func (p *fakePluginNew) HandleNamespaces(namespaces sets.String) error {
+	return fmt.Errorf("not expected")
+}
+func (p *fakePluginNew) Commit() error {
+	return p.err
+}
+
+func TestRouteSecretManager(t *testing.T) {
+
+	scenarios := []struct {
+		name               string
+		route              *routev1.Route
+		secretManager      fake.SecretManager
+		eventType          watch.EventType
+		allow              bool
+		expectedRoute      *routev1.Route
+		expectedEventType  watch.EventType
+		expectedRejections map[string]string
+		expectedError      bool
+	}{
+		{
+			name: "route added without externalCertificate",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{},
+			},
+			eventType: watch.Added,
+			expectedRoute: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{},
+			},
+			expectedEventType:  watch.Added,
+			expectedRejections: map[string]string{},
```

Comment:
Can we modify the test case checks below so that the default value for `expectedRejections` would be fine? This would help us not to specify an empty map each time.

Reply:
Thanks for the suggestion. Updated the test to accommodate default value for `expectedRejections`